samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -86,12 +76,8 @@ private String getHostName() {
     }
   }
 
-  public interface ZkLeaderElectorListener {
-    void onBecomingLeader();
-  }
-
   @Override
-  public boolean tryBecomeLeader() {
+  public boolean tryBecomeLeader(LeaderElectorListener leaderElectorListener) {
```

Comment:
I need to look at this part and may be rewrite it little bit. Will do it as a separate patch.